# aerographene: patch portado sem alteração do conteúdo do diff.
# Origem: https://github.com/imputnet/helium (tag 0.18.3, Chromium 154.0.8037.97)
# Commit: 57a40ad82583d21787cff0500150539ce54c5960
# Caminho original: patches/helium/ui/frame-radius-helper.patch
# Licença: GPL-3.0 (Helium); ver LICENSE do Helium.
#
--- /dev/null
+++ b/chrome/browser/ui/views/helium/frame_corner_radius.h
@@ -0,0 +1,12 @@
+// Copyright 2026 The Helium Authors
+// You can use, redistribute, and/or modify this source code under
+// the terms of the GPL-3.0 license that can be found in the LICENSE file.
+
+#ifndef CHROME_BROWSER_UI_VIEWS_HELIUM_FRAME_CORNER_RADIUS_H_
+#define CHROME_BROWSER_UI_VIEWS_HELIUM_FRAME_CORNER_RADIUS_H_
+
+int GetGenericFrameRadius();
+int GetSystemWindowRadius();
+int GetPaddedFrameRadius();
+
+#endif  // CHROME_BROWSER_UI_VIEWS_HELIUM_FRAME_CORNER_RADIUS_H_
--- /dev/null
+++ b/chrome/browser/ui/views/helium/frame_corner_radius.cc
@@ -0,0 +1,23 @@
+// Copyright 2026 The Helium Authors
+// You can use, redistribute, and/or modify this source code under
+// the terms of the GPL-3.0 license that can be found in the LICENSE file.
+
+#include "chrome/browser/ui/views/helium/frame_corner_radius.h"
+
+#include "build/build_config.h"
+#include "chrome/browser/ui/layout_constants.h"
+
+int GetGenericFrameRadius() {
+  return 8;
+}
+
+#if !BUILDFLAG(IS_MAC)
+int GetSystemWindowRadius() {
+  return 8;
+}
+#endif
+
+int GetPaddedFrameRadius() {
+  return GetSystemWindowRadius() -
+         GetLayoutConstant(LayoutConstant::kHeliumBasePadding);
+}
--- /dev/null
+++ b/chrome/browser/ui/views/helium/frame_corner_radius_mac.mm
@@ -0,0 +1,12 @@
+// Copyright 2026 The Helium Authors
+// You can use, redistribute, and/or modify this source code under
+// the terms of the GPL-3.0 license that can be found in the LICENSE file.
+
+#include "chrome/browser/ui/views/helium/frame_corner_radius.h"
+
+int GetSystemWindowRadius() {
+  if (@available(macOS 26, *)) {
+    return 17;
+  }
+  return 12;
+}
--- a/chrome/browser/ui/BUILD.gn
+++ b/chrome/browser/ui/BUILD.gn
@@ -2342,6 +2342,7 @@ static_library("ui") {
       "views/frame/immersive_mode_controller_mac.mm",
       "views/frame/immersive_mode_overlay_views_mac.h",
       "views/frame/immersive_mode_overlay_views_mac.mm",
+      "views/helium/frame_corner_radius_mac.mm",
     ]
 
     deps += [
@@ -2604,6 +2605,8 @@ static_library("ui") {
       "views/frame/vertical_tab_strip_background_blur_backdrop.h",
       "views/frame/vertical_tab_strip_region_view.cc",
       "views/frame/vertical_tab_strip_region_view.h",
+      "views/helium/frame_corner_radius.cc",
+      "views/helium/frame_corner_radius.h",
       "views/idle_bubble.cc",
       "views/intent_picker_bubble_view.cc",
       "views/intent_picker_bubble_view.h",
